pkg/kubeapiserver/default_storage_factory_builder.go

```diff
@@ -50,3 +54,56 @@ func NewStorageFactory(
 	resourceEncodingConfig = resourceconfig.MergeResourceEncodingConfigs(resourceEncodingConfig, resourceEncodingOverrides)
 	return serverstorage.NewDefaultStorageFactory(storageConfig, defaultMediaType, serializer, resourceEncodingConfig, apiResourceConfig, SpecialDefaultResourcePrefixes), nil
 }
+
+// MergeAPIResourceConfigs merges the given defaultAPIResourceConfig with the given resourceConfigOverrides.
+// Exclude the groups not registered in registry, and check if version is
+// not registered in group, then it will fail.
+func MergeAPIResourceConfigs(
+	defaultAPIResourceConfig *serverstorage.ResourceConfig,
+	resourceConfigOverrides utilflag.ConfigurationMap,
+	registry resourceconfig.GroupVersionRegistry,
+) (*serverstorage.ResourceConfig, error) {
+	resourceConfig := defaultAPIResourceConfig
+
+	// Iterate through all group/version/resource overrides specified in runtimeConfig.
+	for key := range resourceConfigOverrides {
+		tokens := strings.Split(key, "/")
+		if len(tokens) != 3 {
+			continue
+		}
+		groupVersionString := tokens[0] + "/" + tokens[1]
+		resource := tokens[2]
+		groupVersion, err := schema.ParseGroupVersion(groupVersionString)
+		if err != nil {
+			return nil, fmt.Errorf("invalid key %s", key)
+		}
+		// Exclude group not registered into the registry.
+		if !registry.IsRegistered(groupVersion.Group) {
```

Comment:
does this mean that e.g. CRDs are rejected here because they are in another registry? If yes 👍

Reply:
Yes